src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -4065,9 +4070,10 @@ private void CopyFileInfoItem(FileInfo file, string destinationPath, bool force,
 
             if (ShouldProcess(resource, action))
             {
-                // Create destinationPath directory. This will fail if the directory already exists
-                // and Force is not selected.
-                CreateDirectory(destination, false);
+                // Create destinationPath directory and apply attributes of source directory.
+                // This will fail if the directory already exists and Force is not selected.
+                var destDir = CreateDirectory(destination, streamOutput: true);
+                destDir.Attributes = (FileAttributes)sourceDirectoryAttributes;
```

Comment:
I wonder if we can cast to FileAttributes here wait not in definition?

Reply:
I don't understand the question.

Reply:
You cast to byte above and then to FileAttributes here - we could cast to FileAttributes right away.

Reply:
If I try to cast directly to FileAttributes
`var sourceDirectoryAttributes = (FileAttributes)sourceDirectoryInfo["Attributes"];`
I receive
`Copy-Item: Unable to cast object of type 'System.Byte' to type 'System.IO.FileAttributes'.`

Reply:
I think we can:
`var sourceDirectoryAttributes = (FileAttributes)(int)sourceDirectoryInfo["Attributes"];`

Here "int" is default underlying type for Enum types.

Reply:
This is now updated to use int instead of byte.

Reply:
@Joe-Zer0 Please use 
`var sourceDirectoryAttributes = (FileAttributes)(int)sourceDirectoryInfo["Attributes"];`